File: workers.c
#include <stdio.h>
struct Job
{
    int less;
    int exact;
    int more;
};
struct Worker
{
    struct Job job;
    int days;
};
int worker_pay(struct Worker worker){
    if(worker.days<20){
        return worker.days * worker.job.less;
    }else if(worker.days ==20){
        return worker.days*worker.job.exact;
    }else{
        return 20*worker.job.exact+(worker.days-20)*worker.job.more;
    }
}
int main(){
    struct Job A, B, C;
    A.exact=800; A.less=650; A.more=1400;
    B.exact=600; B.less=450; B.more=1200;
    C.exact=500; C.less=250; C.more=1000;
    struct Worker a;
    a.job=C;
    a.days=23;
    printf("%d \n", worker_pay(a));
    return 0;
}

File: workers.py
class Job:
    def __init__(self, exact, less, extra):
        self.exact=exact
        self.less=less
        self.extra=extra

    def pay(self, days):
        if days<20:
            return days*self.less
        elif days==20:
            return days*self.exact
        else :
            return 20*self.exact+(days%20)*self.extra


class Worker:
    def __init__(self, days, job):
        self.days=days
        self.job=job
    
    def salary(self):
        return self.job.pay(self.days)


A= Job(800, 600, 1200)
B=Job(700, 500, 1100)
c=Job(500, 250, 1000)

a=Worker(23, A)
print(a.salary())